docs(recipes): Add Netlify section in Deploying (#15673)
gatsbyjs/gatsby: docs/docs/recipes.md

- Learn about [performance optimization](/docs/performance/)
- Read about [other deployment related topics](/docs/deploying-and-hosting/)

### Deploying to Netlify

Use [`netlify-cli`](https://www.netlify.com/docs/cli/) to deploy your Gatsby application without leaving the command line interface.

#### Prerequisites

- A Gatsby site with a single component `index.js`.

> - [netlify-cli](https://www.npmjs.com/package/netlify-cli) npm package

Author: When you run `netlify login` a new browser is opened to login and authorize "Netlify CLI" application or to create a new account if its needed.

Do you think that a Netlify account is a prerequisite?

- Learn about [performance optimization](/docs/performance/)
- Read about [other deployment related topics](/docs/deploying-and-hosting/)

### Deploying to Netlify

> Use `netlify-cli` to deploy your gatsby application without getting out of command line interface.

Reviewer: A few grammatical suggestions 🙂
```suggestion
Use [`netlify-cli`](https://www.netlify.com/docs/cli/) to deploy your Gatsby application without leaving the command line interface.
```

Author: Thanks a lot!
